src/web/components/SlotBodyEnd.tsx

```diff
@@ -16,60 +22,77 @@ type Props = {
-            ophanComponentId: 'ACQUISITIONS_EPIC',
-            platformId: 'GUARDIAN_WEB',
-            campaignCode: 'gdnwb_copts_memco_remote_epic_test_api',
-            abTestName: 'remote_epic_test',
-            abTestVariant: 'api',
-            referrerUrl: window?.location.origin + window?.location.pathname
-        },
-        localisation: {
-            countryCode: 'GB', // TODO: make this dynamic
-        },
-        targeting: {
-            contentType,
-            sectionName,
-            shouldHideReaderRevenue,
-            isMinuteArticle,
-            isPaidContent,
-            tags,
-        },
-    };
+    const [epicContent, setEpicContent] = useState<{ html: string; css: string}>();
 
-    const endpointUrl = 'https://contributions.guardianapis.com/epic';
-    const { data: responseBody, error } = useApi(endpointUrl, {
-        method: 'POST',
-        headers: {
-            'Content-Type': 'application/json',
-        },
-        body: JSON.stringify(contributionsPayload),
-    });
+    useEffect(() => {
+        const callFetch = async () => {
+            // Putting together the request payload
+            const contributionsPayload = {
+                tracking: {
+                    ophanPageId: window?.guardian.config.ophan.pageViewId,
+                    ophanComponentId: 'ACQUISITIONS_EPIC',
+                    platformId: 'GUARDIAN_WEB',
+                    campaignCode: 'gdnwb_copts_memco_remote_epic_test_api',
+                    abTestName: 'remote_epic_test',
+                    abTestVariant: 'api',
+                    referrerUrl: window?.location.origin + window?.location.pathname
+                },
+                localisation: {
+                    countryCode
+                },
+                targeting: {
+                    contentType,
+                    sectionName,
+                    shouldHideReaderRevenue,
+                    isMinuteArticle,
+                    isPaidContent,
+                    tags,
+                    showSupportMessaging: shouldShowSupportMessaging(),
+                    isRecurringContributor: isRecurringContributor(isSignedIn),
+                    lastOneOffContributionDate: getLastOneOffContributionDate(),
+                },
+            };
 
-    if (error) {
-        window.guardian.modules.sentry.reportError(error, 'slot-body-end');
-        return null;
-    }
-
-    if (responseBody && responseBody.data) {
-        const { data } = responseBody;
+            const getEpicContent = () => {
+                const endpointUrl = 'https://contributions.guardianapis.com/epic';
+                return fetch(endpointUrl, {
+                    method: 'POST',
+                    headers: {
+                        'Content-Type': 'application/json',
+                    },
+                    body: JSON.stringify(contributionsPayload)
+                })
+                .then((response) => {
+                    if (!response.ok) {
```

Comment:
If the response is non-OK here, should we return a failed promise and add the error handling (i.e. Sentry reporting) to a `catch` at the end (or a `try`/`catch` around `await getEpicContent()`). That would have the added benefit of handling any other failure scenarios (for example bad json returned, network error).

Reply:
Yeah, we're now sending errors to Sentry, but by making use of `useApi`, then wrapping the Sentry call in a condition that checks for any errors.

Reply:
RE extracting the `undefined` checks into separate functions, my preference would be to keep this really simple and I think the functions, despite making the condition more readable, will add unnecessary noise to this file. What we're doing here is really simple, both variables get initialised as `undefined` via `useState()` and subsequently get assigned `true` or `false`. We want to wait until that happens before we render the component. I know it's unusual to check for `boolean !== undefined`, hence the confusion around this, but this code is semantic and compact.